socksio/socks5.py

```diff
@@ -189,30 +219,28 @@ def __init__(self) -> None:
     def state(self) -> SOCKS5State:
         return self._state
 
-    def authenticate(self, methods: typing.List[SOCKS5AuthMethod]) -> None:
-        auth_request = SOCKS5AuthRequest(methods)
-        self._data_to_send += auth_request.dumps()
+    @singledispatchmethod
```

Comment:
This is clever, I didn't know about `singledispatchmethod`!

Reply:
Yeah, it makes things more readable for very clear cases like this 🙂